Feature/27/prepare scatter for gradient colorscale

asizemore opened the pull request:
Addresses issue #27 

This PR allows the overlay variable passed to `scattergl.dt` to be continuous. 

## Rules for continuous overlay variables:
- No trend lines. `scattergl.dt` will err if passed a continuous overlay var and `value != 'raw`. 
- Continuous overlay vars operate as one group. In other words, there is one row in the resulting data.table per panel, not per overlay var and panel (as is the case otherwise).The overlay var is returned as a list that should have the same length as `'seriesX'` and `'seriesY'`.

## Testing
New tests added to the scattergl context. All relevant tests pass.

## Notes
- Instead of being ordered, `c(group, panel, 'seriesX', 'seriesY') `, the returned pd data.table will have columns `c(panel, 'seriesX', 'seriesY', group)`. I do not expect this to be a problem but can reorder if that is not the case!
- Decided to keep the overlay var validation step in validateScatterPD in case we for some reason are send an overlay var with `dataShape=zebra`, etc.

d-callan commented:
it occurs to me that probably we want to call this thing something like 'seriesGradientColorscale' or similar. and then in utils-json.R when we write out the variable details, include a check for a continuous overlay variable and write that out similarly to the x and y vars. it does allow for the overlay var details to be in two different places in the response depending on its data shape, but i think it might be ok.

asizemore commented:
latest updates include:

- renamed column to 'seriesGradientColorscale'
- `addStrataVariableDetails` ignores overlay if overlay$dataShape is continuous
- if overlay$dataShape is continuous, overlay written to 'config'
- reorganized a bit of `getJSON` so that in config the order will be x, y, z, overlay instead of overlay, x, y, z.
- added tests for appropriately formatted json

asizemore commented:
Closes #27
